ci/build-and-test-software.yml

```diff
@@ -0,0 +1,33 @@
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+# These tasks build and test the OpenTitan software using `meson`.
+
+parameters:
+- name: extra_meson_init_args
+  displayName: "Extra Arguments for ./meson_init.sh"
+  type: string
+  default: ""
+- name: allowed_to_fail
+  displayName: "Is this build allowed to fail?"
+  type: string # This doesn't work as a boolean, of course
+  default: 'false'
+- name: perform_tests
+  displayName: "Should this build run tests?"
```

Comment:
Please don't use quotes unless necessary (that's the style mcy used consistently throughout our pipelines config)